sql/sys_vars.cc

```diff
@@ -3463,15 +3463,21 @@ static Sys_var_mybool Sys_timed_mutexes(
        DEPRECATED(""));
 
 static char *server_version_ptr;
-static Sys_var_charptr Sys_version(
+static Sys_var_version Sys_version(
        "version", "Server version",
        READ_ONLY GLOBAL_VAR(server_version_ptr), NO_CMD_LINE,
-       IN_SYSTEM_CHARSET, DEFAULT(server_version));
+       IN_SYSTEM_CHARSET, DEFAULT(MYSQL_SERVER_VERSION));
+
+static char *server_version_suffix_ptr;
+static Sys_var_charptr Sys_version_suffix(
+       "version_suffix", "version_suffix",
+       GLOBAL_VAR(server_version_suffix_ptr), NO_CMD_LINE,
```

Comment:
What's the size limit of `version_suffix`?
What happens if its more than `SERVER_VERSION_LENGTH`?

Reply:
`version_suffix` has no limits but now `version` have limit of `SERVER_VERSION_LENGTH` and only first `SERVER_VERSION_LENGTH - 1 - sizeof(MYSQL_SERVER_VERSION)` bytes of `version_suffix` will be concatenated what is checked in:
```
    if (suffix_ptr_len + version_ptr_len + 1 > SERVER_VERSION_LENGTH)
      suffix_ptr_len = SERVER_VERSION_LENGTH - version_ptr_len - 1;
```